packages/eas-build-job/src/metadata.ts

```diff
@@ -149,6 +149,11 @@ export type Metadata = {
    * Indicates whether this is (likely, we can't be 100% sure) development client build.
    */
   developmentClient?: boolean;
+
+  /**
+   * Which package manager will be used for the build. Determined based on lockfiles in the project directory.
+   */
+  requiredPackageManager?: 'npm' | 'pnpm' | 'yarn' | 'bun';
```

Comment:
I decided to call it `requiredPackageManager` because this is the package manager that must get installed at the "install custom tools" step if not already, and if it isn't, we should fail the build.